public/apps/configuration/panels/role-view/role-view.tsx

```diff
@@ -22,33 +22,152 @@ import {
   EuiSpacer,
   EuiTabbedContent,
   EuiTitle,
+  EuiPageContent,
+  EuiText,
+  EuiLink,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPageBody,
+  EuiInMemoryTable,
+  EuiEmptyPrompt,
 } from '@elastic/eui';
 import { BreadcrumbsPageDependencies } from '../../../types';
 import { buildHashUrl } from '../../utils/url-builder';
 import { ResourceType, Action } from '../../types';
-
-const tabs = [
-  {
-    id: 'permissions',
-    name: 'Permissions',
-    disabled: false,
-    content: <Fragment>Permission working in progress</Fragment>,
-  },
-  {
-    id: 'users',
-    name: 'Users',
-    disabled: false,
-    content: <Fragment>Users working in progress</Fragment>,
-  },
-];
+import { getRoleMappingDeta, MappedUsersListing } from '../../utils/role-mapping-utils';
 
 interface RoleViewProps extends BreadcrumbsPageDependencies {
   roleName: string;
 }
 
+function getmappedUserColumns() {
+  return [
+    {
+      field: 'user_type',
+      name: 'User type',
+      sortable: true,
+    },
+    {
+      field: 'user_name',
+      name: 'User',
+      sortable: true,
+      truncateText: true,
+    },
+  ];
+}
+
 export function RoleView(props: RoleViewProps) {
   const duplicateRoleLink = buildHashUrl(ResourceType.roles, Action.duplicate, props.roleName);
 
+  const [mappedUsers, setMappedUsers] = useState<MappedUsersListing[]>([]);
+  const [errorFlag, setErrorFlag] = useState(false);
+  const [selection, setSelection] = useState<MappedUsersListing[]>([]);
+  const [message, setMessage] = useState(
```

Comment:
Why need a state to manage message here?

Reply:
I removed state. We don't need here.